Spigot-API-Patches/0039-LootTable-API.patch

```diff
@@ -0,0 +1,303 @@
+From 69514393965a5983f8c9dd1050da581ad8337cd3 Mon Sep 17 00:00:00 2001
+From: Aikar <aikar@aikar.co>
+Date: Sun, 1 May 2016 15:19:49 -0400
+Subject: [PATCH] LootTable API
+
+Provides API to control what Loot Table an object uses.
+
+Also provides an Event to control if a lootable inventory should
+auto replenish for a player.
+
+Provides methods to determine players looted state for an object
+
+diff --git a/src/main/java/com/destroystokyo/paper/loottable/Lootable.java b/src/main/java/com/destroystokyo/paper/loottable/Lootable.java
+new file mode 100644
+index 0000000..94f7d4c
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/loottable/Lootable.java
+@@ -0,0 +1,44 @@
++package com.destroystokyo.paper.loottable;
++
++public interface Lootable {
++
++    /**
++     * Gets the name of the Loot Table to be used in the World Folder
++     * @return The name, or null if no loot table exists
++     */
++    String getLootTableName();
++
++    /**
++     * Returns whether or not this object has a Loot Table
++     * @return Has a loot table
++     */
++    default boolean hasLootTable() {
++        return getLootTableName() != null;
++    }
++
++    /**
++     * Sets the name of the Loot Table to be used in the World Folder
++     * Will use a random seed (0)
++     *
++     * @param name name in either foo or minecraft:foo format
++     * @return The previous Loot Table before the change
++     */
++    default String setLootTable(String name) {
++        return setLootTable(name, 0);
++    }
++
++    /**
++     * Sets the name of the Loot Table to be used in the World Folder
++     * Uses supplied Seed
```

Comment:
There has to be some expectation of the user knowing what a Loot Table is and how they work. But maybe ill add a generic message on the class itself, as each method does not need to document how minecraft implements Loot Tables